src/main/java/net/mcreator/element/GeneratableElement.java

```diff
@@ -101,14 +103,25 @@ public void setLastModElement(ModElement lastModElement) {
 		@Override
 		public GeneratableElement deserialize(JsonElement jsonElement, Type type,
 				JsonDeserializationContext jsonDeserializationContext) throws JsonParseException {
-			ModElementType modElementType = jsonDeserializationContext
-					.deserialize(jsonElement.getAsJsonObject().get("_type"), ModElementType.class);
+			String newType = jsonElement.getAsJsonObject().get("_type").getAsString();
+			switch (newType) {
+			case "gun":
+				newType = "rangeditem";
+				break;
+			case "mob":
+				newType = "livingentity";
+				break;
+			}
+			jsonElement.getAsJsonObject().addProperty("_type", newType);

```

Comment:
I don't think it makes sense to add property here as it won't be used in deserialization anymore

Reply:
I’m not familiar with the generator system yet, so I thought it would be useful for unused mod elements (so mod elements that aren’t saved again by the user).

Reply:
You were thinking correct, but this is not saved if you set it here. No need to save though as your code here will fix this each time until user edits it to parentally "fix" it